src/core_plugins/kibana/server/lib/export/collect_index_patterns.js

```diff
@@ -0,0 +1,23 @@
+export default function collectIndexPatterns(savedObjectsClient, panels) {
+  const docs = panels.reduce((acc, panel) => {
+    const { kibanaSavedObjectMeta, savedSearchId } = panel.attributes;
+
+    if (kibanaSavedObjectMeta && !savedSearchId) {
+      const searchSource = JSON.parse(kibanaSavedObjectMeta.searchSourceJSON);
```

Comment:
We should probably check that the `kibanaSavedObjectMeta` has a `searchSourceJSON` property right? Do we want to add `try/catch` around this parsing?